draft-ietf-quic-http.md

```diff
@@ -1390,28 +1390,34 @@ for more details.
 
 An HTTP implementation MUST NOT send frames or requests which would be invalid
 based on its current understanding of the peer's settings.  All settings begin
-at an initial value, and are updated upon receipt of a SETTINGS frame.  For
-servers, the initial value of each client setting is the default value.
+at an initial value.  Each endpoint can use these values to send messages before
+the peer's SETTINGS frame has arrived.  This removes the need to wait for the
+SETTINGS frame before sending messages.  When the SETTINGS frame arrives, any
+settings are changed to their new values.
+
+For servers, the initial value of each client setting is the default value.
 
 For clients using a 1-RTT QUIC connection, the initial value of each server
-setting is the default value. When a 0-RTT QUIC connection is being used, the
+setting is the default value.  When a 0-RTT QUIC connection is being used, the
 initial value of each server setting is the value used in the previous session.
-Clients MUST store the settings the server provided in the session being resumed
-and MUST comply with stored settings until the current server settings are
-received.  A client can use these initial values to send requests before the
-server's SETTINGS frame has arrived.  This removes the need for a client to wait
-for the SETTINGS frame before sending requests.
+Clients SHOULD store the settings the server provided in the connection where
+resumption information was provided. A client MUST comply with either stored
+settings -- or default values, if no values are stored -- when attempting 0-RTT.
+Once a server has provided new settings, clients MUST comply with those values.
 
 A server can remember the settings that it advertised, or store an
 integrity-protected copy of the values in the ticket and recover the information
 when accepting 0-RTT data. A server uses the HTTP/3 settings values in
-determining whether to accept 0-RTT data.
+determining whether to accept 0-RTT data.  If the server cannot determine that
+the settings remembered by a client are compatible with its current settings, it
+MUST NOT accept 0-RTT data.  (Remembered settings are compatible if a client
+complying with those settings would not violate the server's current settings.)
```

Comment:
(I expect our implementation will just require an exact byte-for-byte match. It saves adding a funny callback and I wouldn't expect the values to actually change much. Same for the QUIC transport parameters.)

Reply:
Storing a version number in place of storing settings is entirely appropriate.

Reply:
Sure, that's a plausible optimization. Although that's orthogonal to whether you believe in equality or a weaker comparison.

(Whether the optimization is worth it, I'm not sure. The benefit is slightly smaller tickets. The cost is you need to make sure the meaning of your version number is consistent, which is a hidden sharp edge to deployment.)